webapp/src/components/SingleStudy/XpansionView/index.tsx

```diff
@@ -0,0 +1,282 @@
+  const [settings, setSettings] = useState<XpansionSettings>();
+  const [candidates, setCandidates] = useState<Array<XpansionCandidate>>();
+  const [constraints, setConstraints] = useState<Array<string>>();
+  const [links, setLinks] = useState<Array<LinkCreationInfo>>();
+  const [capacities, setCapacities] = useState<Array<string>>();
+  const [createConfigView, setCreateConfigView] = useState<boolean>(false);
+  const [loaded, setLoaded] = useState<boolean>(false);
+  const [singleConstraint, setSingleConstraint] = useState<{filename: string; content: string}>();
+  const [singleCapa, setSingleCapa] = useState<{filename: string; content: MatrixType}>();
+
+  const init = useCallback(async (after: () => void = () => { /* noop */ }) => {
+    try {
+      const exist = await xpansionConfigurationExist(study.id);
+      if (exist) {
+        const tempSettings = await getXpansionSettings(study.id);
+        setSettings(tempSettings);
+        const tempCandidates = await getAllCandidates(study.id);
+        for (let i = 0; i < tempCandidates.length; i += 1) {
+          tempCandidates[i].link = tempCandidates.map((item) => item.link.split(' - ').map((index) => transformNameToId(index)).join(' - '))[i];
+        }
+        setCandidates(tempCandidates);
+        const tempConstraints = await getAllConstraints(study.id);
+        setConstraints(tempConstraints);
+        const tempLinks = await getAllLinks(study.id);
+        setLinks(tempLinks);
+        const tempCapa = await getAllCapacities(study.id);
+        setCapacities(tempCapa);
+      } else {
+        setCreateConfigView(true);
+      }
+      if (after) {
+        after();
+      }
+    } catch (e) {
+      enqueueErrorSnackbar(enqueueSnackbar, t('xpansion:xpansionError'), e as AxiosError);
+    } finally {
+      setLoaded(true);
+    }
+  }, [study.id, enqueueSnackbar, t]);
+
+  const createXpansion = async () => {
+    try {
+      await createXpansionConfiguration(study.id);
+    } catch (e) {
+      enqueueErrorSnackbar(enqueueSnackbar, t('xpansion:createXpansionError'), e as AxiosError);
+    } finally {
+      setCreateConfigView(false);
+      setLoaded(true);
+      init();
+    }
+  };
+
+  const deleteXpansion = async () => {
+    try {
+      await deleteXpansionConfiguration(study.id);
+    } catch (e) {
+      enqueueErrorSnackbar(enqueueSnackbar, t('xpansion:deleteXpansionError'), e as AxiosError);
+    } finally {
+      setCreateConfigView(true);
+    }
+  };
+
+  const createCandidate = async (name: string, link: string) => {
+    try {
+      await addCandidate(study.id, { name, link, 'annual-cost-per-mw': 0, 'max-investment': 0 });
+      setOpenModal(false);
+    } catch (e) {
+      enqueueErrorSnackbar(enqueueSnackbar, t('xpansion:createCandidateError'), e as AxiosError);
+    } finally {
+      init(() => setSelectedItem({ name, link, 'annual-cost-per-mw': 0, 'max-investment': 0 }));
```

Comment:
well maybe split the init function into sub functions so that you won't fetch all the xpansion config, but just the candidates